velox/expression/CastExpr.cpp

```diff
@@ -645,6 +644,7 @@ void CastExpr::evalSpecialForm(
   stats_.numProcessedRows += rows.countSelected();
   auto timer = cpuWallTimer();
   apply(rows, input, context, fromType, toType, result);
+  context.releaseVector(input);
```

Comment:
Nice. 

nit: Perhaps, add a small comment to explain that we are returning 'input' to the vector pool so it can be reused.